baseplate/lib/secrets.py

```diff
@@ -495,8 +495,11 @@ def __init__(
         backoff: Optional[float] = None,
     ):  # pylint: disable=super-init-not-called
         self.path = Path(path)
+        if not self.path.is_dir():
+            raise ValueError(f"{self.path} must be a directory")
```

Comment:
Can we say "root of the CSI volume?" I know it's a little verbose but I think it'll be a helpful pointer

Reply:
This might be my go showing but I very much like formatting error messages as "Expected {path} to be a directory, was a {type-of-thing-it-is}"

Reply:
I think I covered both of these - unless I'm missing something, I don't think we can only ask "is it type X" and not "what type is it", so it's hard to show the {type-of-thing-it-is}